### ISSUE TITLE
"Warning: fwrite(): 49 is not a valid stream resource" when running tests from eclipse PDT

### ISSUE THREAD

**Issue by troosan** [sebastianbergmann/phpunit#2944]
When I'm running my tests from PDT (version 5) I'm getting the following warning.

```
Warning: fwrite(): 49 is not a valid stream resource in .../vendor/phpunit/phpunit/src/Util/Printer.php on line 100

Call Stack:
    0.0002     237344   1. {main}() .../vendor/phpunit/phpunit/phpunit:0
    0.0077    1311048   2. PHPUnit_TextUI_Command::main() .../vendor/phpunit/phpunit/phpunit:52
    0.0077    1311784   3. PHPUnit_TextUI_Command->run() .../vendor/phpunit/phpunit/src/TextUI/Command.php:116
    0.0252    3816536   4. PHPUnit_TextUI_TestRunner->doRun() .../vendor/phpunit/phpunit/src/TextUI/Command.php:186
   13.0712   27742432   5. PHPUnit_Util_Printer->write() .../vendor/phpunit/phpunit/src/TextUI/TestRunner.php:580
   13.0712   27742480   6. fwrite() .../vendor/phpunit/phpunit/src/Util/Printer.php:100
```

https://github.com/sebastianbergmann/phpunit/blob/4acfe0a4b973002e5b3f9b07f6d8fe9dfea75dad/src/Util/Printer.php#L101-L102

I did not check in detail why `$this->out` is not a resource but changing the condition to 

```php
        if (is_resource($this->out)) {
```
fixes ("hides" is probably the better term) the issue.

I'm using phpunit 5.7.26 (not tried with version 6), running with php 5.6 (I did not check with php 7.0)

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by troosan** [comment]
@sebastianbergmann indeed, I fully understand, but it will be pretty difficult to reproduce outside of eclipse I guess.
I have the feeling it is linked to whether code coverage is being produced or not.

I'll try to it narrow down a bit.

**Comment by sebastianbergmann** [comment]
I do not use Eclipse so I cannot help with debugging this issue, sorry.

**Comment by troosan** [comment]
@sebastianbergmann then again, checking with is_resource() is probably safer than just checking for not null

**Comment by troosan** [comment]
@sebastianbergmann This could explain it.
https://stackoverflow.com/questions/2488770/gettype-and-unknown-type-in-php

I guess the phpunit eclipse plugin is closing the stream, so when the printer is trying to print to it it's already closed.

Anyway, the bug described above is solved in PHP 7.2
